### ISSUE TITLE
Make Power Flow Direction Load to Grid depending on positive or negative value of essential_power

### ISSUE THREAD

**Issue by tttooommmeeekkk** [slipx06/sunsynk-power-flow-card#491]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

The flowing dot actually flows in the direction to the house (load) independent of positive or negative values of the entity "essential_power". I have installed an plug-socket-pv-generator as an auxillary. If this supplys more power to the load, than the house actually requires, the rest is loaded to the grid and the value of the "essential_power" entity turns into a negative value. The flowing dot actually does not turn the direction.

### Expected behaviour

I expect a flowing dot from/to load in dependency of the essential_power entity-value.

### Possible Solutions

If the value is positive (mine is inverted - positive means, the house does not need all the power from the plug-socket-pv-generator) the dot should flow from right to left.
If the value is negative (house needs more power, than supplied by the plug-socket-pv-generator) the dot should flow from left to right.
Maybe make it invertible too.

### Mode

full

### Context / Reason

The flowing-directions are an essential visualization style of your great implementation and it should be great to have the right flow-direction also be implemented at the load, so that I see, where the power is directed to (from the house/load or to the house/load).

![power_flow-issue_sunsyk-power-flow-card](https://github.com/slipx06/sunsynk-power-flow-card/assets/151519742/3b522aa1-19dd-49ff-9453-a441f1da25f2)

**Comment by slipx06** [comment]
Sue I can add that. There is already an option to invert the load

```
load:
  invert_load: false
```

**Comment by tttooommmeeekkk** [comment]
I guess you misunderstood my suggestion.
I know, that there is an option to invert the value.
What I suggested is to invert the direction of the moving dot in reference to the value. The dot actually always moves from left to right no matter if the value is positive or negative.

**Comment by slipx06** [comment]
Please update to v4.44.0 and test

**Comment by tttooommmeeekkk** [comment]
Fantastic. It works (direction changes).
Is there a possibility of having the dots moving from right to left, if the value is positive amd from left to right, when the value is negative as shown in my screenshot (actually the coded logic its the other way around).

**Comment by slipx06** [comment]
On the load side positive power is interpreted as consumption and negative as production.  If your sensor is inverted you would need to use the `invert_load: true` option

**Comment by tttooommmeeekkk** [comment]
I recognized this.
But I still question if this is logic:

- If power is sent to grid it is a consumption of energy (by my neighbours), the power is negative and the dots flow from left to right (into the grid).

- If power is sent to the load/house it is equally a consumption of energy (even though by myself), but you say the power must be positive to make the dots flow from left to right (into the load/house).

In my opinion it is a logic error, don't you think so?!
(I.e. compared to the "tesla-flow-card" implementation there a negative load flows into the grid.)

Please do not misunderstand my annoying comments: Your work has so many fantastic features and i love it. Maybe you consider about my suggestion on a boring winterday...

Best regards

**Comment by slipx06** [comment]
Its a valid point and their are some inconsistencies in the card.   Perhaps this quote can put things into perspective

> The practice of "positive" and "negative" power to reference "import" and "export" power is purely by convention, and most importantly based on the desired measurement result.  For example, in most buildings, power from the grid, is considered "import" (positive), and power that is pushed to the grid is "export" (negative).  However, in a solar inverter application, the inverter may be considered the source, and it is desirable for its power delivered to be considered positive.

Its really a matter of preference and manufacturers don't apply this consistently. The work around to set `show_absolute: true` and then power reading are always positive and you refer to the animation for flow direction. I don't have that setting for the load section as I never considered that it would be producing energy. If you think it would be useful let me know and I can add it

**Comment by tttooommmeeekkk** [comment]
Thanks for your accomodation.
Actually I have solved the problem with a possible negative load_power by creating a new sensor that accumulates the load_power with the power of the plug-socket-pv-generator so that the load_power_gross never becomes negative and the essential_load always shows the real load (gross) of the whole house.

But the `show_absolute: true` option would nevertheless be a nice work-around in my opinion if someone has a similar configuration and wants to see the flow in both directions (corresponding to the pos./neg. prefix of the load). I actually refraind from this option and put the plug-socket-pv-generator back to the solar-entity (away from aux-entity) because I got problems with the autarky-values (see screenshot im my first posting).
